src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -508,8 +528,25 @@ private FlowControlException InvokeHelper()
             return flowControlException;
         }
 
-        // NTRAID#Windows Out Of Band Releases-915506-2005/09/09
-        // Removed HandleUnexpectedExceptions infrastructure
+#if !UNIX
+        /// <summary>
+        /// Invokes the InvokeThreadProc() method on new thread, and flows calling thread 
+        /// impersonation as needed.
+        /// </summary>
+        private void InvokeThreadProcImpersonate()
+        {
+            if (_identityToImpersonate != null)
+            {
```

Comment:
This check doesn't seem to be needed. When using this method, we already know `_identityToImpersonate` is not null because `Utils.TryGetWindowsImpersonatedIdentity(out _identityToImpersonate)` returns true.

Reply:
Good point.  I forgot to remove this when refactoring...